src/representations/representation_manager_spherical_expansion.hh

```diff
@@ -0,0 +1,535 @@
+      stream << "Soap vector size " << this->get_feature_size() << std::endl;
+      for (size_t radial_n{0}; radial_n < this->max_radial; ++radial_n) {
+        stream << "n = " << radial_n << std::endl;
+        stream << this->soap_vectors[center].row(radial_n) << std::endl;
+      }
+    }
+
+    std::vector<precision_t>& get_representation_raw_data() {
+      return this->soap_vectors.get_raw_data();
+    }
+
+    size_t get_feature_size() {
+      // (should be) equivalent:
+      //return this->n_species * this->max_radial
+      //    * pow(this->max_angular + 1, 2);
+      return this->soap_vectors.get_nb_comp();
+    }
+
+    size_t get_center_size() {
+      return this->soap_vectors.get_nb_item();
+    }
+
+    /**
+     * Return whether the radial sigmas and overlap matrix have been precomputed
+     *
+     * This only needs to be done once on initialization, but it is not done in
+     * the constructor to avoid spending time and possibly throwing exceptions
+     * there.  Instead, by default, it is done once on the first call of the
+     * compute() method and the precomputed results are saved for subsequent
+     * calls.
+     */
+    bool get_is_precomputed() {
+      return this->is_precomputed;
+    }
+
+   protected:
+   private:
+    double interaction_cutoff{};
+    double cutoff_smooth_width{};
+    size_t max_radial{};
+    size_t max_angular{};
+    size_t n_species{};
+    std::string gaussian_sigma_str{};
+    // TODO(max-veit) these are specific to the radial Gaussian basis
+    Eigen::VectorXd radial_sigmas{};
+    Eigen::VectorXd radial_norm_factors{};
+    Eigen::VectorXd radial_sigma_factors{};
+    Eigen::MatrixXd radial_nl_factors{};
+    Eigen::MatrixXd radial_ortho_matrix{};
+    bool is_precomputed{false};
+
+    Manager_t& structure_manager;
+    Property_t soap_vectors;
+    internal::GaussianSigmaType gaussian_sigma_type{};
+
+    hypers_t hypers{};
+  };
+
+
+  /** Compute common prefactors for the radial Gaussian basis functions */
+  template<class Mngr>
+  void RepresentationManagerSphericalExpansion<Mngr>::
+      precompute_radial_sigmas() {
+    using std::pow;
+    size_t radial_n;
+    size_t angular_l;
+
+    for (radial_n = 0; radial_n < this->max_radial; radial_n += 1) {
+      this->radial_sigmas[radial_n] = std::max(
+            std::sqrt(static_cast<double>(radial_n)), 1.0)
+          * this->interaction_cutoff / static_cast<double>(this->max_radial);
+    }
+
+    // Precompute common prefactors
+    for (radial_n = 0; radial_n < this->max_radial; radial_n++) {
+      this->radial_norm_factors(radial_n) = std::sqrt(
+          2.0 / std::tgamma(1.5 + radial_n)
+          * pow(this->radial_sigmas[radial_n], 3.0 + 2.0*radial_n));
+      for (angular_l = 0; angular_l < this->max_angular + 1; angular_l++) {
+        this->radial_nl_factors(radial_n, angular_l) =
+            std::exp2(-0.5 * (1.0 + angular_l - radial_n))
```

Comment:
I looked it up; it's fine in this case, since the integers are converted to the floating-point type (although I should verify that it's `double` and not `float`)

Reply:
To clarify, this works because floating-point literals in C++ default to type `double` (cf. the -0.5 or 1.0 in the expression above). It might still be nice to make this explicit with a nice static cast.